velox/common/memory/Memory.h

```diff
@@ -138,10 +138,20 @@ class MemoryManager {
 
   ~MemoryManager();
 
-  /// Tries to get the singleton memory manager. If not previously initialized,
-  /// the process singleton manager will be initialized.
-  FOLLY_EXPORT static MemoryManager& getInstance(
-      const MemoryManagerOptions& options = MemoryManagerOptions{});
+  /// Creates the process-wide default memory manager based on the specified
+  /// 'options'. This function throws if the default memory manager has already
+  /// been set.
+  static void setInstance(const MemoryManagerOptions& options);
+
+  /// Returns the process-wide default memory manager instance. This function
+  /// throws if the default memory manager has not been set.
+  static MemoryManager& getInstance();
+
+  /// Returns the process-wide default memory manager instance if exists,
+  /// otherwise creates one based on the specified 'options'. Deprecates after
```

Comment:
typo: Deprecates -> Deprecated. Do not use. Remove once existing call sites are updated.